moshi/src/main/java/com/squareup/moshi/ArrayJsonAdapter.kt

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.moshi
+
+import java.lang.reflect.Type
+
+/**
+ * Converts arrays to JSON arrays containing their converted contents. This supports both primitive
+ * and object arrays.
+ */
+internal class ArrayJsonAdapter(
+  private val elementClass: Class<*>,
+  private val elementAdapter: JsonAdapter<Any>
+) : JsonAdapter<Any?>() {
+  override fun fromJson(reader: JsonReader): Any {
+    val list = buildList<Any?> {
```

Comment:
We don't currently support null values in an array/list. Please null check the returned value for fromJson and throw a JsonDataException indicating an unexpected null element at the current reader path

Reply:
Are you sure? This changes the behavior on [this test](https://github.com/square/moshi/blob/master/moshi/src/test/java/com/squareup/moshi/AdapterMethodsTest.java#L472) that previously added null elements to the list.

Reply:
interesting, I guess we can just leave this for now then